preview/Win7Msix/Win7MSIXInstaller/InstallUI.cpp

```diff
@@ -145,8 +145,37 @@ LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam)
                 }
             }
             break;
+            case IDC_LAUNCHBUTTON:
+                ui->LaunchInstalledApp();
+                break;
         }
         break;
+    case WM_INSTALLCOMPLETE_MSG:
+    {
+        DestroyWindow(g_CancelbuttonHWnd);
+        ui->CreateLaunchButton(hWnd, windowRect);
+        UpdateWindow(hWnd);
+        ShowWindow(g_progressHWnd, SW_HIDE); //hide progress bar
+        ShowWindow(g_checkboxHWnd, SW_HIDE); //hide launch check box
+        if (g_launchCheckBoxState) {
+            ui->LaunchInstalledApp(); // launch app
+            DestroyWindow(hWnd); // close msix app installer
+        }
+        else
+        {
+            //wait for user to click launch button or close the window
+            while (true)
+            {
+                MsgWaitForMultipleObjects(0, NULL, FALSE, INFINITE, QS_ALLINPUT);

```

Comment:
>MsgWaitForMultipleObjects [](start = 16, length = 25)

even though you're waiting infinitely for 0 handles, you should still check the return value. It should only ever return one value, but you should still check it. Also the open brace (for the while) should be on a line by itself